### ISSUE TITLE
Signed email has signature not attached (but within the mail)

### ISSUE THREAD

**Issue by sisu4u** [PHPMailer/PHPMailer#476]
Hi,

I use the following code to sign and send an email:
```
$mail = new PHPMailer(true); 
$mail->CharSet = 'utf-8';
$mail->AddAddress("me@example.com");
$mail->SetFrom("me@example.com");
$mail->Subject = "TestMail";
$mail->Body = "This is a test mail";
$mail->sign('cert.crt', 'cert.key', 'xxx');
$mail->Send();
```
I generated a certificate with StartSSL. Then I exported the certificate from my chrome (pfx).
I used the following commands to generate the crt and key file:
```
openssl pkcs12 -in crt.pfx -nocerts -out cert.key -nodes
openssl pkcs12 -in crt.pfx -nokeys -out cert.crt
```

The email is sent without any error. However, the signature is not attached as a file as supposed.
This is how the email is shown:
```
MIME-Version: 1.0
Content-Type: multipart/signed; protocol="application/x-pkcs7-signature"; micalg="sha-256"; boundary="----F4131BBA0FF79046CC5FD7B59A3AEABF"

This is an S/MIME signed message

------F4131BBA0FF79046CC5FD7B59A3AEABF
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

This is a test mail

------F4131BBA0FF79046CC5FD7B59A3AEABF
Content-Type: application/x-pkcs7-signature; name="smime.p7s"
Content-Transfer-Encoding: base64
Content-Disposition: attachment; filename="smime.p7s"

MIIJFwYJKoZIhvcNAQcCoIIJCDCCCQQCAQExDzANBglghkgBZQMEAgEFADALBgkq
hkiG9w0BBwGgggY0MIIGMDCCBRigAwIBAgIDDx04MA0GCSqGSIb3DQEBCwUAMIGM
MQswCQYDVQQGEwJJTDEWMBQGA1UEChMNU3RhcnRDb20gTHRkLjErMCkGA1UECxMi
U2VjdXJlIERpZ2l0YWwgQ2VydGlmaWNhdGUgU2lnbmluZzE4MDYGA1UEAxMvU3Rh
cnRDb20gQ2xhc3MgMSBQcmltYXJ5IEludGVybWVkaWF0ZSBDbGllbnQgQ0EwHhcN
MTUwODIxMjIwNDEyWhcNMTYwODIxMjAyNzM5WjBIMR8wHQYDVQQDDBZyZWNobnVu
Z0BzZmctc2luZ2VuLmRlMSUwIwYJKoZIhvcNAQkBFhZyZWNobnVuZ0BzZmctc2lu
Z2VuLmRlMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnWd9L08KiXM3
J129FGfFiJkB60y/9VvRbC3oBD6oYCLyCBNgkXlm/DkyK6bEtaE64iI2NfMuR8E9
6PqqNmlZDexlRmOTcUneIwR8Ds5jt3GMxXvxPbYyjAC8q1AD4TLCtXcHEzRyodjo
/3E4yQgxlm1jYq14PqVWGn1rAWxhQy8hzIokGtifHmLQ5bqfMJRDJ9AEXO2NnCkC
ewnVfEYKuktMe0fWPbopu7fVVbrcgvPjaldk37ERyoLAjol5sxT4AhU8HGXo2k3L
17T05DgZV7xdaJp8xg3n9FRcfHn/O4qFcQR6P4yJmUY+6q2P7ds60jqcvIreZ5ki
cwMHlthhZwIDAQABo4IC3DCCAtgwCQYDVR0TBAIwADALBgNVHQ8EBAMCBLAwHQYD
VR0lBBYwFAYIKwYBBQUHAwIGCCsGAQUFBwMEMB0GA1UdDgQWBBTLR+y8Y/LIFnrU
j0i7mycBNZBGOTAfBgNVHSMEGDAWgBRTcu2SnODaywFcfH6WNU7y1LhRgjAhBgNV
HREEGjAYgRZyZWNobnVuZ0BzZmctc2luZ2VuLmRlMIIBTAYDVR0gBIIBQzCCAT8w
ggE7BgsrBgEEAYG1NwECAzCCASowLgYIKwYBBQUHAgEWImh0dHA6Ly93d3cuc3Rh
cnRzc2wuY29tL3BvbGljeS5wZGYwgfcGCCsGAQUFBwICMIHqMCcWIFN0YXJ0Q29t
IENlcnRpZmljYXRpb24gQXV0aG9yaXR5MAMCAQEagb5UaGlzIGNlcnRpZmljYXRl
IHdhcyBpc3N1ZWQgYWNjb3JkaW5nIHRvIHRoZSBDbGFzcyAxIFZhbGlkYXRpb24g
cmVxdWlyZW1lbnRzIG9mIHRoZSBTdGFydENvbSBDQSBwb2xpY3ksIHJlbGlhbmNl
IG9ubHkgZm9yIHRoZSBpbnRlbmRlZCBwdXJwb3NlIGluIGNvbXBsaWFuY2Ugb2Yg
dGhlIHJlbHlpbmcgcGFydHkgb2JsaWdhdGlvbnMuMDYGA1UdHwQvMC0wK6ApoCeG
JWh0dHA6Ly9jcmwuc3RhcnRzc2wuY29tL2NydHUxLWNybC5jcmwwgY4GCCsGAQUF
BwEBBIGBMH8wOQYIKwYBBQUHMAGGLWh0dHA6Ly9vY3NwLnN0YXJ0c3NsLmNvbS9z
dWIvY2xhc3MxL2NsaWVudC9jYTBCBggrBgEFBQcwAoY2aHR0cDovL2FpYS5zdGFy
dHNzbC5jb20vY2VydHMvc3ViLmNsYXNzMS5jbGllbnQuY2EuY3J0MCMGA1UdEgQc
MBqGGGh0dHA6Ly93d3cuc3RhcnRzc2wuY29tLzANBgkqhkiG9w0BAQsFAAOCAQEA
ump/JaM01R6zd9yHJZcCgpH4AOlHO4Mao0XL3NTUI6a/Febit0PZo2jo7zHDWFxF
OOYptUfupfTGciiPwcAK+TYNvZ+CCevF7OZMqbK7uDz0UEITTmECeiKFPxpeNb8E
QISCvb6wqzVleU3vunlNZvm3SuUCt7W48k9vBxY9qQUZLpSd2R/EuAv6H3fA/4Ru
7gU+WxTlFfoYZU2FUDh/fE+i6h/FfunlaCJ4Z7k1KYE2vvG+Q2jyacrFpgpLyvL6
UNvovfgx9IYilu7TcpHWxCjiYz5ZgCZXMNZwjg6FIIcQXO4alfNT3Pg/ZjwGAdEE
84Us8DAPkB5WVTI8MY0Y4zGCAqcwggKjAgEBMIGUMIGMMQswCQYDVQQGEwJJTDEW
MBQGA1UEChMNU3RhcnRDb20gTHRkLjErMCkGA1UECxMiU2VjdXJlIERpZ2l0YWwg
Q2VydGlmaWNhdGUgU2lnbmluZzE4MDYGA1UEAxMvU3RhcnRDb20gQ2xhc3MgMSBQ
cmltYXJ5IEludGVybWVkaWF0ZSBDbGllbnQgQ0ECAw8dODANBglghkgBZQMEAgEF
AKCB5DAYBgkqhkiG9w0BCQMxCwYJKoZIhvcNAQcBMBwGCSqGSIb3DQEJBTEPFw0x
NTA4MjIxNzU5MDVaMC8GCSqGSIb3DQEJBDEiBCCXD0DexB6kaB1trpZBPkf22Sio
at7ubDQMJI2BqAWQijB5BgkqhkiG9w0BCQ8xbDBqMAsGCWCGSAFlAwQBKjALBglg
hkgBZQMEARYwCwYJYIZIAWUDBAECMAoGCCqGSIb3DQMHMA4GCCqGSIb3DQMCAgIA
gDANBggqhkiG9w0DAgIBQDAHBgUrDgMCBzANBggqhkiG9w0DAgIBKDANBgkqhkiG
9w0BAQEFAASCAQBsx4XGV2IKDHY59HJzK5gxQXiO40v3DT/hJaegWd9e8QD3g9l+
8e6jTW16F+zr7OmrrT1oieZczWwGUNn/U8lj8NMGvvOm6W3Zo/K8gXww4BJSZ5Ir
gvrZWYEz0zvubGyh9r+6NwZCSTn/G2q1cb1tKfsnCl2URALP1txqKDjz0nZ6f27W
nhdocufa8IhQfbFtPUk6LQP55WCoVVNkkA+jQd8LSNoNHs6yw4tugtG8IkTPkShe
U3JO511Bv2xrqjv2pggn0ms2GLJ6zaMfTqnENd60JiAVBVBG7at2OM6Of/fvnrRR
3ikz5OQdZ0AR7Ip026wLZCRH1H4mYdYMhX2G

------F4131BBA0FF79046CC5FD7B59A3AEABF--
```

The complete source of the mail looks like this:
```
Return-Path: <me@example.com>
Delivered-To: me@example.com
Received: by xxx (Postfix, from userid 33)
	id 7B3F18E0D2E; Sat, 22 Aug 2015 19:59:05 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=example.com; s=mail;
	t=1440266345; bh=M28v979upBkc1S+ueKfcFx/TidmJoT4KHtcS83p/Cj8=;
	h=To:Subject:Date:From:From;
	b=liWD7Zti5VneMPtqbi7RPEOcet2b7HXy1LTzOKSm+HafoE7QVh5lIwWxzlYCYKzdX
	 fqmW15lAktTMOoxq2+kIDWp31hQ7a/sXwC4+O042hf48NEAq46KlYvmt+Z3u6Xc6te
	 I7FgmxBNoIslFwlTrD7NJQzoX5nmUS7gpoKtIxuw=
To: me@example.com
Subject: TestMail
Date: Sat, 22 Aug 2015 19:59:05 +0200
From: me@example.com
Message-ID: <9a4ae05b6daca551ff667556b2e8f046@xxx>
X-Priority: 3
X-Mailer: PHPMailer 5.2.9 (https://github.com/PHPMailer/PHPMailer/)

MIME-Version: 1.0
Content-Type: multipart/signed; protocol="application/x-pkcs7-signature"; micalg="sha-256"; boundary="----F4131BBA0FF79046CC5FD7B59A3AEABF"

This is an S/MIME signed message

------F4131BBA0FF79046CC5FD7B59A3AEABF
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

This is a test mail

------F4131BBA0FF79046CC5FD7B59A3AEABF
Content-Type: application/x-pkcs7-signature; name="smime.p7s"
Content-Transfer-Encoding: base64
Content-Disposition: attachment; filename="smime.p7s"

MIIJFwYJKoZIhvcNAQcCoIIJCDCCCQQCAQExDzANBglghkgBZQMEAgEFADALBgkq
hkiG9w0BBwGgggY0MIIGMDCCBRigAwIBAgIDDx04MA0GCSqGSIb3DQEBCwUAMIGM
MQswCQYDVQQGEwJJTDEWMBQGA1UEChMNU3RhcnRDb20gTHRkLjErMCkGA1UECxMi
U2VjdXJlIERpZ2l0YWwgQ2VydGlmaWNhdGUgU2lnbmluZzE4MDYGA1UEAxMvU3Rh
cnRDb20gQ2xhc3MgMSBQcmltYXJ5IEludGVybWVkaWF0ZSBDbGllbnQgQ0EwHhcN
MTUwODIxMjIwNDEyWhcNMTYwODIxMjAyNzM5WjBIMR8wHQYDVQQDDBZyZWNobnVu
Z0BzZmctc2luZ2VuLmRlMSUwIwYJKoZIhvcNAQkBFhZyZWNobnVuZ0BzZmctc2lu
Z2VuLmRlMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnWd9L08KiXM3
J129FGfFiJkB60y/9VvRbC3oBD6oYCLyCBNgkXlm/DkyK6bEtaE64iI2NfMuR8E9
6PqqNmlZDexlRmOTcUneIwR8Ds5jt3GMxXvxPbYyjAC8q1AD4TLCtXcHEzRyodjo
/3E4yQgxlm1jYq14PqVWGn1rAWxhQy8hzIokGtifHmLQ5bqfMJRDJ9AEXO2NnCkC
ewnVfEYKuktMe0fWPbopu7fVVbrcgvPjaldk37ERyoLAjol5sxT4AhU8HGXo2k3L
17T05DgZV7xdaJp8xg3n9FRcfHn/O4qFcQR6P4yJmUY+6q2P7ds60jqcvIreZ5ki
cwMHlthhZwIDAQABo4IC3DCCAtgwCQYDVR0TBAIwADALBgNVHQ8EBAMCBLAwHQYD
VR0lBBYwFAYIKwYBBQUHAwIGCCsGAQUFBwMEMB0GA1UdDgQWBBTLR+y8Y/LIFnrU
j0i7mycBNZBGOTAfBgNVHSMEGDAWgBRTcu2SnODaywFcfH6WNU7y1LhRgjAhBgNV
HREEGjAYgRZyZWNobnVuZ0BzZmctc2luZ2VuLmRlMIIBTAYDVR0gBIIBQzCCAT8w
ggE7BgsrBgEEAYG1NwECAzCCASowLgYIKwYBBQUHAgEWImh0dHA6Ly93d3cuc3Rh
cnRzc2wuY29tL3BvbGljeS5wZGYwgfcGCCsGAQUFBwICMIHqMCcWIFN0YXJ0Q29t
IENlcnRpZmljYXRpb24gQXV0aG9yaXR5MAMCAQEagb5UaGlzIGNlcnRpZmljYXRl
IHdhcyBpc3N1ZWQgYWNjb3JkaW5nIHRvIHRoZSBDbGFzcyAxIFZhbGlkYXRpb24g
cmVxdWlyZW1lbnRzIG9mIHRoZSBTdGFydENvbSBDQSBwb2xpY3ksIHJlbGlhbmNl
IG9ubHkgZm9yIHRoZSBpbnRlbmRlZCBwdXJwb3NlIGluIGNvbXBsaWFuY2Ugb2Yg
dGhlIHJlbHlpbmcgcGFydHkgb2JsaWdhdGlvbnMuMDYGA1UdHwQvMC0wK6ApoCeG
JWh0dHA6Ly9jcmwuc3RhcnRzc2wuY29tL2NydHUxLWNybC5jcmwwgY4GCCsGAQUF
BwEBBIGBMH8wOQYIKwYBBQUHMAGGLWh0dHA6Ly9vY3NwLnN0YXJ0c3NsLmNvbS9z
dWIvY2xhc3MxL2NsaWVudC9jYTBCBggrBgEFBQcwAoY2aHR0cDovL2FpYS5zdGFy
dHNzbC5jb20vY2VydHMvc3ViLmNsYXNzMS5jbGllbnQuY2EuY3J0MCMGA1UdEgQc
MBqGGGh0dHA6Ly93d3cuc3RhcnRzc2wuY29tLzANBgkqhkiG9w0BAQsFAAOCAQEA
ump/JaM01R6zd9yHJZcCgpH4AOlHO4Mao0XL3NTUI6a/Febit0PZo2jo7zHDWFxF
OOYptUfupfTGciiPwcAK+TYNvZ+CCevF7OZMqbK7uDz0UEITTmECeiKFPxpeNb8E
QISCvb6wqzVleU3vunlNZvm3SuUCt7W48k9vBxY9qQUZLpSd2R/EuAv6H3fA/4Ru
7gU+WxTlFfoYZU2FUDh/fE+i6h/FfunlaCJ4Z7k1KYE2vvG+Q2jyacrFpgpLyvL6
UNvovfgx9IYilu7TcpHWxCjiYz5ZgCZXMNZwjg6FIIcQXO4alfNT3Pg/ZjwGAdEE
84Us8DAPkB5WVTI8MY0Y4zGCAqcwggKjAgEBMIGUMIGMMQswCQYDVQQGEwJJTDEW
MBQGA1UEChMNU3RhcnRDb20gTHRkLjErMCkGA1UECxMiU2VjdXJlIERpZ2l0YWwg
Q2VydGlmaWNhdGUgU2lnbmluZzE4MDYGA1UEAxMvU3RhcnRDb20gQ2xhc3MgMSBQ
cmltYXJ5IEludGVybWVkaWF0ZSBDbGllbnQgQ0ECAw8dODANBglghkgBZQMEAgEF
AKCB5DAYBgkqhkiG9w0BCQMxCwYJKoZIhvcNAQcBMBwGCSqGSIb3DQEJBTEPFw0x
NTA4MjIxNzU5MDVaMC8GCSqGSIb3DQEJBDEiBCCXD0DexB6kaB1trpZBPkf22Sio
at7ubDQMJI2BqAWQijB5BgkqhkiG9w0BCQ8xbDBqMAsGCWCGSAFlAwQBKjALBglg
hkgBZQMEARYwCwYJYIZIAWUDBAECMAoGCCqGSIb3DQMHMA4GCCqGSIb3DQMCAgIA
gDANBggqhkiG9w0DAgIBQDAHBgUrDgMCBzANBggqhkiG9w0DAgIBKDANBgkqhkiG
9w0BAQEFAASCAQBsx4XGV2IKDHY59HJzK5gxQXiO40v3DT/hJaegWd9e8QD3g9l+
8e6jTW16F+zr7OmrrT1oieZczWwGUNn/U8lj8NMGvvOm6W3Zo/K8gXww4BJSZ5Ir
gvrZWYEz0zvubGyh9r+6NwZCSTn/G2q1cb1tKfsnCl2URALP1txqKDjz0nZ6f27W
nhdocufa8IhQfbFtPUk6LQP55WCoVVNkkA+jQd8LSNoNHs6yw4tugtG8IkTPkShe
U3JO511Bv2xrqjv2pggn0ms2GLJ6zaMfTqnENd60JiAVBVBG7at2OM6Of/fvnrRR
3ikz5OQdZ0AR7Ip026wLZCRH1H4mYdYMhX2G

------F4131BBA0FF79046CC5FD7B59A3AEABF--
```

The DKIM is added by my postfix. Sending signed emails usually works (for example from Thunderbird or SOGo).

Also a complete example would be appreciated I just found very old examples which seem not to apply to the recent versions any more.

**Comment by Synchro** [comment]
It's probably that blank line before the `MIME-Version: 1.0` header line - I think that was fixed in 5.2.10, so give that a try. There is an example of using S/MIME signing in the unit tests, though we could do with a better example for the examples folder - care to write one?

**Comment by sisu4u** [comment]
Ohh and yes it got fixed by updating to 5.2.10. Thanks.
